src/modules/launcher/Plugins/Microsoft.Plugin.Folder/Sources/QueryInternalDirectory.cs

```diff
@@ -0,0 +1,189 @@
+        };
+
+        private static string _warningIconPath;
+
+        public QueryInternalDirectory(FolderSettings folderSettings, IQueryFileSystemInfo queryFileSystemInfo)
+        {
+            _settings = folderSettings;
+            _queryFileSystemInfo = queryFileSystemInfo;
+        }
+
+        private static bool HasSpecialChars(string search)
+        {
+            return search.Any(c => SpecialSearchChars.Contains(c));
+        }
+
+        public static SearchOption GetSearchOptions(string query)
+        {
+            // give the ability to search all folder when it contains a >
+            if (query.Any(c => c.Equals('>')))
+            {
+                return SearchOption.AllDirectories;
+            }
+
+            return SearchOption.TopDirectoryOnly;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Globalization", "CA1308:Normalize strings to uppercase", Justification = "Do not want to change the behavior of the application, but want to enforce static analysis")]
+        private (string search, string incompleteName) Process(string search)
+        {
+            string incompleteName = string.Empty;
+            if (HasSpecialChars(search) || !_queryFileSystemInfo.Exists($@"{search}\"))
+            {
+                // if folder doesn't exist, we want to take the last part and use it afterwards to help the user
+                // find the right folder.
+                int index = search.LastIndexOf('\\');
+
+                // No slashes found, so probably not a folder
+                if (index <= 0 || index >= search.Length - 1)
+                {
+                    return default;
+                }
+
+                // Remove everything after the last \ and add *
+                incompleteName = search.Substring(index + 1)
+                    .Replace(">", string.Empty, StringComparison.Ordinal)
+                    .ToLower(CultureInfo.InvariantCulture) + "*";
+                search = search.Substring(0, index + 1);
+                if (!_queryFileSystemInfo.Exists(search))
+                {
+                    return default;
+                }
+            }
+            else
+            {
+                // folder exist, add \ at the end of doesn't exist
+                if (!search.EndsWith(@"\", StringComparison.InvariantCulture))
+                {
+                    search += @"\";
+                }
+            }
+
+            return (search, incompleteName);
+        }
+
+        public IEnumerable<IItemResult> Query(string querySearch)
+        {
+            if (querySearch == null)
+            {
+                throw new ArgumentNullException(nameof(querySearch));
+            }
+
+            var processed = Process(querySearch);
+
+            if (processed == default)
+            {
+                yield break;
+            }
+
+            var (search, incompleteName) = processed;
+            var searchOption = GetSearchOptions(querySearch);
```

Comment:
@P-Storm.  With regards to [this](https://github.com/microsoft/PowerToys/pull/6600#discussion_r492884897) comment.  If you change `querySearch` to 'incompleteName`.  You should be able to use your original implementation of `StartsWith('>'` and maintain the existing query functionality.   Please let me know if I'm missing something.

Reply:
I have tested the change that you propose. This change doesn't matter and I reverted it. I have tried the following examples and these are currently the same implementations:

* `>C:\DwarfFortress` this maps a normal map, the `>` is an actionword, not a search.
* `C:\>DwarfFortress\>` this is an invalid folder (the first `>`) so no results
* `C:\>DwarfFortress\` this is an invalid folder (path contains `>`) so no results
* `C:\DwarfFortress` this is a valid recursive search.

This is currently with the latest code of mine.